photoframe/source/app/src/main/java/org/kaaproject/kaa/demo/photoframe/activities/DevicesActivity.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2014-2016 CyberVision, Inc.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaaproject.kaa.demo.photoframe.activities;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+import org.kaaproject.kaa.demo.photoframe.DeviceInfo;
+import org.kaaproject.kaa.demo.photoframe.R;
+import org.kaaproject.kaa.demo.photoframe.adapter.DevicesAdapter;
+import org.kaaproject.kaa.demo.photoframe.communication.Events;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DevicesActivity extends BaseActivity {
+
+
+    private TextView mNoData;
+    private ListView mDevices;
+    private DevicesAdapter mAdapter;
+
+    private List<DeviceInfo> mDeviceInfos = new ArrayList<>();
+
+    public static void start(Context context) {
```

Comment:
Activity here instead of context.
you cannot start activity like this from application or service